client/src/components/App/App.js

```diff
@@ -1,21 +1,90 @@
 import React, { Component } from 'react';
 import './App.css';
 import styled from 'styled-components';
-import NavBar from '../NavBar';
+let password = '',
+    confirmPassword ='';
 
 
 class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      signUpEmail: '',
+      signUpPassword:'',
+      signUpPasswordConfirm:''
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.validateInput = this.validateInput.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleConfirmPasswordChange = this.handleConfirmPasswordChange.bind(this);
+  }
+
   render() {
     return (
       <div>     
           Sign Up <br/>
-          <input
-              type="text"
-              placeholder="First Name"/><br/> 
-          <input type="submit" value="Submit" />
+          <form>
+            <input
+              type="email"
+              name="signUpEmail"
+              placeholder="Email"
+              value={this.state.signUpEmail}
+              onChange={this.handleChange}/><br/> 
+             <input
+              type="password"
+              name="signUpPassword"
+              placeholder="Password"
+              value={this.state.signUpPassword}
+              onChange={this.handlePasswordChange}/><br/>
+            <input
+              type="password"
+              name="signUpPasswordConfirm"
+              placeholder="Confirm Password"
+              value={this.state.signUpPasswordConfirm}
+              onChange={this.handleConfirmPasswordChange}/><br/>  
+            <input 
+              type="submit" 
+              value="Submit" 
+              onClick={this.validateInput}/>
+          </form> 
       </div>
     );
   }
-}
 
+  handleChange(event){
```

Comment:
@amtadros I think if this is a Create React App configured repo you should be able to define this method as a class property that holds a reference to a function instead of a class method. That should mean you won't have to call `this.handleChange = this.handleChange.bind(this)` in the constructor.

It would look something like `handleChange = (event) => {...`

Let me know if you have any questions about that syntax!

Reply:
@blaketarter I changed the syntax of the functions. Let me know if this is what you're looking for! Also, let me know if I should have fixed both changes you mentioned before pushing again :)